apps/protocol-api/src/prisma/resolvers/ActivityTypes.ts

```diff
@@ -0,0 +1,59 @@
+import * as TypeGraphQL from 'type-graphql';
+import { ActivityType } from '../generated/type-graphql';
+import { Context } from './types';
+
+@TypeGraphQL.InputType('GetActivityTypesPerUserAndDAOs')
+export class GetActivityTypesPerUserAndDAOs {
+  @TypeGraphQL.Field(_type => Number)
+  userId: number;
+}
+
+@TypeGraphQL.ArgsType()
+export class GetActivityTypesPerUserAndDAOsArgs {
+  @TypeGraphQL.Field(_type => GetActivityTypesPerUserAndDAOs, {
+    nullable: false,
+  })
+  where: GetActivityTypesPerUserAndDAOs;
+}
+
+@TypeGraphQL.ObjectType()
+export class ActivityTypesByUser {
+  @TypeGraphQL.Field(_type => Number)
+  id: number;
+
+  @TypeGraphQL.Field(_type => Boolean)
+  active: boolean;
+
+  @TypeGraphQL.Field(_type => String)
+  name: string;
+
+  @TypeGraphQL.Field(_type => String)
+  createdAt: string;
+
+  @TypeGraphQL.Field(_type => String)
+  updatedAt: string;
+}
+
+@TypeGraphQL.Resolver(_of => ActivityType)
+export class ActivityTypesCustomResolver {
+  @TypeGraphQL.Query(_returns => [ActivityTypesByUser], {
+    nullable: false,
+  })
+  async getActivityTypesByUser(
+    @TypeGraphQL.Ctx() { prisma }: Context,
+    @TypeGraphQL.Args() args: GetActivityTypesPerUserAndDAOsArgs,
+  ) {
+    const user_id = args.where.userId;
+
+    return await prisma.$queryRaw<ActivityTypesByUser>`
+      SELECT at.*, ua.user_id, gat.guild_id
+      FROM "ActivityType" at
+             LEFT JOIN "UserActivity" ua
+                       ON at.id = ua.activity_type_id
+             LEFT JOIN "GuildActivityType" gat
+                       ON at.id = gat.activity_type_id
+      WHERE ua.user_id = ${user_id}
+         OR (gat.guild_id IN (SELECT guild_id FROM "GuildUser" WHERE user_id = ${user_id}))
+      ORDER BY at.name;`;
```

Comment:
This can be replaced by something like 

```

    prisma.activityType.findMany({
      OR: [
        {
          users: {
            some: {
              user_id: { equals: 1 },
            },
          },
        },
        {
          guilds: {
            some: {
              guild_id: { equals: 1 },
            },
          },
        },
      ],
    });
```